src/main.c

```diff
@@ -58,7 +58,8 @@ static int	init_shell(char **envp)
 		return (EXIT_FAILURE);
 	if (init_job_control_stuff() == EXIT_FAILURE)
 		return (EXIT_FAILURE);
-	init_history_stuff();
+	if (init_history_stuff() == EXIT_FAILURE)
+		exit(0);
```

Comment:
exit shouldnt be 0, it has to be different than 0 because of error, can u change it to 1 and i will approve this change